lib/puppet/type/docker_compose.rb

```diff
@@ -0,0 +1,18 @@
+Puppet::Type.newtype(:docker_compose) do
+    @doc = "Runs docker-compose"
+
+    ensurable
+
+    newparam(:name) do
```

Comment:
Given this isn't used for anything else in the provider I would probably make this the file path to the compose.yml file, which removes the need for the additional source property.